### PR TITLE
Add registration of Check in employees attendance functionality

### PR DESCRIPTION
**Pull request by jestradadeveloper**
# Description
This Commit introduce a set of changes that allows the employee give his private number and be able to register his check in. Changes made by this PR are:
 - Creation of EmployeesRegistration controller to manage check in and check outs 
 - Implementation of post routes to give react check in component, the chance of making requests with Axios library and retrieve checkin information
 - Changes to the Check in react component just implementing the check in button
 - Changes to Employee controller to take off the json renders statements , to prevent repetitive code and because we don't need the json render in that controller.
 - Changes to Employee and Attendance model for manage the message state and give the check in functionality if there is no attendance for an employee then this model will create one for him, if the employee already done check in an attendance then will give to the user the right message to notify him is already registered a check in.

  ## Screenshots
Captura de Pantalla 2021-10-27 a la(s) 9 51 47

### CODE REVIEW

File: app/controllers/attendances_controller.rb [hunk 1]
```diff
@@ -64,8 +74,16 @@ def set_attendance
       @attendance = Attendance.find(params[:id])
     end
 
+    def set_employee
+      @employee = Employee.find_by private_number: params[:private_number]
```
Comment (jclopezdev) [inline app/controllers/attendances_controller.rb:78]: If you add an exclamation mark here `!` the find_by will throw an `ActiveRecord::RecordNotFound` error, so you don't need to add the unless present..., what do you think?

File: app/controllers/attendances_controller.rb [hunk 2]
```diff
@@ -64,8 +74,16 @@ def set_attendance
...
+      unless @employee.present?
+        return render json: {"message": "The user with these private number does not exist"}, status: :unprocessable_entity
+      end
+    end
+
     # Only allow a list of trusted parameters through.
     def attendance_params
       params.require(:attendance).permit(:check_in, :check_out, :employee_id, :user_id)
     end
+
```
Comment (jclopezdev) [inline app/controllers/attendances_controller.rb:88]: Please remove these unnecessary extra line

**Review by jclopezdev — APPROVED** [review]
I left some simple comments and it looks better now, you still need to improve more the specs and organize better the code, but right now I'm OOO, so if you can not check something with the other mentors please feel free to merge it and continue working on the next tickets 👍

File: app/controllers/attendances_controller.rb (hunk 1 above)
Comment (jestradadeveloper) [reply app/controllers/attendances_controller.rb:78]: I didn't know i will test this, thanks

File: app/controllers/attendances_controller.rb (hunk 2 above)
Comment (jestradadeveloper) [reply app/controllers/attendances_controller.rb:88]: i think this extra line at the end of every file is added by default with an extension, someone mentioned before, it was a good practice add it, it isn't?.

**Review by JuanequeX — APPROVED** [review]
LGTM

**Review by CUBOFIG — APPROVED** [review]
LGTM!

**Pull request merged**